packages/java/src/components/Enum.tsx

```diff
@@ -37,9 +42,15 @@ export interface EnumMemberProps extends ObjectModifiers {
 }
 
 export function EnumMember(props: EnumMemberProps) {
+  const enumScope = useScope() as JavaEnumScope;
+
   const collectedArgs = collectArguments(props.arguments);
   const args = props.arguments ? code`(${collectedArgs})` : "";
   const name = useJavaNamePolicy().getName(props.name, "enum-member");
 
-  return code`${name}${args}`;
+  // Handle position in enum scope to determine delimiter
+  enumScope.members.push(name);
+  const delimiter = memo(() => enumScope.getDelimiter(name));
```

Comment:
Use `mapJoin` in the enum when members are provided as opposed to handling delimiter in the enum member element. Try to keep the scope dedicated to symbol tracking information.

Reply:
No worries, I've reverted the change